### PR TITLE
Introduce new iex helpers

### PR DESCRIPTION
**Pull request by tonini**

### CODE REVIEW

**Comment by whatyouhide** [comment]
In alchemist, this will only work at compile-time right?

**Comment by tonini** [comment]
@whatyouhide I think not, it works on runtime. Have a look at the emacs interactive functions which calls `t/1` it's nothing special at all.

**Comment by whatyouhide** [comment]
@tonini I meant, will it work if I have something like this:

```elixir
name = "Samuel"
```

and I run `i/1` on `name` instead of a literal like `"Samuel"`?

**Comment by tonini** [comment]
@whatyouhide Ah I see :) no, It can't work that way :)

But you can do just the following:

```elixir
name = "Samuel"
IEx.Helpers.i name
```

and just `alchemist-eval-buffer`

**Comment by whatyouhide** [comment]
Sure ok, I just wanted to make sure I got it right :) Thanks for clarifying!

**Comment by tonini** [comment]
@whatyouhide you're always right mate :) But I always need some time to get it right ;-)

File: alchemist-server/test/server_test.exs [hunk 1]
```diff
@@ -92,6 +92,32 @@ defmodule ServerTest do
     """
   end
 
+  # The IEx.Helpers.t and IEx.Helpers.i are functionality which come with
+  # Elixir version 1.2.0
+  if Version.match?(System.version, ">=1.2.0-rc.0") do
```
Comment (lexmag) [inline alchemist-server/test/server_test.exs]: `">=1.2.0-rc"` for brevity?

File: alchemist-server/lib/api/info.exs [hunk 2]
```diff
@@ -42,8 +44,46 @@ defmodule Alchemist.API.Info do
...
   def normalize(request) do
-    {{_, name }, _} = Code.eval_string(request)
-    name
+    try do
+      arguments = Code.eval_string(request)
+      case arguments do
+        {{_, type }, _}     -> type
+        {{_, type, arg}, _} ->
+          IO.puts System.version
+          if Version.match?(System.version, ">=1.2.0-rc.0") do
```
Comment (lexmag) [inline alchemist-server/lib/api/info.exs]: `">=1.2.0-rc"` for brevity? :smile:

File: alchemist-server/lib/api/info.exs [hunk 3]
```diff
@@ -42,8 +44,46 @@ defmodule Alchemist.API.Info do
...
+  end
+
+  def process(nil) do
+    IO.puts "END-OF-INFO"
+  end
+
   def normalize(request) do
-    {{_, name }, _} = Code.eval_string(request)
-    name
+    try do
```
Comment (lexmag) [inline alchemist-server/lib/api/info.exs]: I think we could rewrite to clearly separate "falling" part from regular code:
```elixir
try do
  Code.eval_string(request)
rescue
  _e -> nil
else
  {{_, type }, _} -> type
  {{_, type, arg}, _} -> ...
end
```

Comment (tonini) [inline alchemist-server/lib/api/info.exs]: Looks great, I will make a PR on https://github.com/tonini/alchemist-server :)

Thanks :+1:

**Pull request closed without merging**